numpy/fft/fftpack.py

```diff
@@ -86,7 +92,8 @@ def _raw_fft(a, n=None, axis=-1, init_function=fftpack.cffti,
     # As soon as we put wsave back into the cache, another thread could pick it
     # up and start using it, so we must not do this until after we're
     # completely done using it ourselves.
-    fft_cache[n].append(wsave)
+    with _cache_lock:
+        fft_cache[n].append(wsave)
```

Comment:
Well, it depends on using the index `[n]` in `_raw_fft` which calls `__getitem__`, which is where all the good stuff happens. That is rather convoluted IMHO, I think it would be clearer to call `setitem` and have all the goodstuff in one spot. Maybe `cache_item` instead of `setitem`, but that is minor.